pkg/storage/util_test.go

```diff
@@ -29,8 +31,8 @@ import (
 )
 
 var (
-	fooLabelsWithName = labels.Labels{{Name: "foo", Value: "bar"}, {Name: "__name__", Value: "logs"}}
-	fooLabels         = labels.Labels{{Name: "foo", Value: "bar"}}
+	fooLabelsWithNameAndShard = labels.Labels{{Name: "foo", Value: "bar"}, {Name: "__name__", Value: "logs"}, {Name: distributor.ShardLbName, Value: "3"}}
```

Comment:
concern: it is great that you could cover the behavior when `__stream_shard` is present but by changing this, you actually did change all other scenarios so that we're not covering when **stream_shard isn't present**. Just a thought.

Reply:
My thought was that `__stream_shard__` should be removed the same way `__name__` is because it's not ever relevant to a user. I'm not sure, though and happy to hear from anyone else.

Reply:
hmmm fair point, `__stream_shard` behavior is definitely very similar to `__name__`. That said, I'm approving it.